internal/resolver/resolver.go

```diff
@@ -58,42 +58,51 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 	locked := make([]*chart.Dependency, len(reqs))
 	missing := []string{}
 	for i, d := range reqs {
-		constraint, err := semver.NewConstraint(d.Version)
-		if err != nil {
-			return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
-		}
+		var constraint *semver.Constraints
+		var chartpath string
+		var err error
 
-		if d.Repository == "" {
-			// Local chart subfolder
-			if _, err := GetLocalPath(filepath.Join("charts", d.Name), r.chartpath); err != nil {
-				return nil, err
-			}
-
-			locked[i] = &chart.Dependency{
-				Name:       d.Name,
-				Repository: "",
-				Version:    d.Version,
+		// If version is defined
+		if d.Version != "" {
+			constraint, err = semver.NewConstraint(d.Version)
+			if err != nil {
+				return nil, errors.Wrapf(err, "dependency %q has an invalid version/constraint format", d.Name)
 			}
-			continue
 		}
-		if strings.HasPrefix(d.Repository, "file://") {
-			chartpath, err := GetLocalPath(d.Repository, r.chartpath)
-			if err != nil {
-				return nil, err
+
+		// Local chart
+		if d.Repository == "" || strings.HasPrefix(d.Repository, "file://") {
+			if d.Repository == "" {
```

Comment:
this conditional IMHO has made the code more complex. The empty repository and `file://` loading logic has been combined into a single branch. With the intent of utilizing the same version processing logic.

But... (while this sounds like a good thing) the subtle differences in these code paths is leading to lots of conditional checks. It would be great to find a way to simplify the conditional logic here I think.

Reply:
I understand, but I don't think it's as bad as it sounds. We know that in real life helm will never have to iterate on these conditions long enough to make a difference in performance. It's not like there will be a chart with thousands of dependencies (which is probably still a low number of iterations).

Nevertheless, I'm still thinking about it.